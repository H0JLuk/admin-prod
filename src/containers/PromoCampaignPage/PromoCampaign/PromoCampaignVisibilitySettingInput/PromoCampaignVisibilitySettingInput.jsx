import React from 'react';
import cn from 'classnames';
import { Form, Switch } from 'antd';
import AutocompleteLocationAndSalePoint from '../../../../components/Form/AutocompleteLocationAndSalePoint/AutocompleteLocationAndSalePoint';
import { getStringOptionValue } from '../../../../utils/utils';

import styles from './PromoCampaignVisibilitySettingInput.module.css';

const VISIBILITY_FIELD = {
    name: 'visibility',
    label: 'Включить видимость',
};

const FORM_NAME = 'createVisibilityForm';

const PromoCampaignVisibilitySettingInput = ({
    error,
    onLocationChange,
    onSalePointChange,
    visibility,
    onVisibilityChange,
    location,
    salePoint,
    columnMode,
}) => (
    <div className={ cn(styles.wrapper, { [styles.columnWrapper]: columnMode }) }>
        <Form
            className={ cn(styles.form, { [styles.columnForm]: columnMode }) }
            name={ FORM_NAME }
            requiredMark={ false }
        >
            <AutocompleteLocationAndSalePoint
                onLocationChange={ onLocationChange }
                onSalePointChange={ onSalePointChange }
                locationLabelClassNames="required"
                locationId={ location?.id }
                initialLocationValue={ getStringOptionValue(location || undefined) }
                initialSalePointValue={ salePoint?.name ?? '' }
                error={ error }
                columnMode={ columnMode }
            />
            <div className={ styles.visibility }>
                <label htmlFor={ VISIBILITY_FIELD.name } className={ styles.label }>
                    { VISIBILITY_FIELD.label }
                </label>
                <Switch
                    className={ cn(styles.switcher, { [styles.switch]: visibility }) }
                    checked={ visibility }
                    onChange={ onVisibilityChange }
                    id={ VISIBILITY_FIELD.name }
                />
            </div>
        </Form>
        { error.server && <div className={ styles.error }>{ error.server } </div> }
    </div>
);

export default PromoCampaignVisibilitySettingInput;
