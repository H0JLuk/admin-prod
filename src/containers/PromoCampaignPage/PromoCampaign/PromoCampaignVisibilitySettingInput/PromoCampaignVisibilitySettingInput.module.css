.wrapper {
    padding: 25px 30px;
    background-color: #f8f8f8;
}

form.form {
    display: flex;
    align-items: flex-start;
}

form.form > div:nth-child(2) {
    margin-right: 10px;
}

.label {
    min-width: fit-content;
    margin-right: 10px;
    cursor: pointer;
    user-select: none;
}

.switch:global(.ant-switch-checked) {
    background-color: #09a552;
}

.visibility {
    flex-basis: 30%;
    margin-top: 35px;
}
