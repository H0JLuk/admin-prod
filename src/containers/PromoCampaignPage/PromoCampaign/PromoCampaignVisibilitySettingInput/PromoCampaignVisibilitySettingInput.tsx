import React from 'react';
import { LocationDto, SalePointDto } from '@types';
import cn from 'classnames';
import { Switch } from 'antd';
import AutocompleteLocationAndSalePoint from '@components/Form/AutocompleteLocationAndSalePoint/AutocompleteLocationAndSalePoint';
import { getStringOptionValue } from '@utils/utils';

import styles from './PromoCampaignVisibilitySettingInput.module.css';

export type PromoCampaignVisibilitySettingInputProps = {
    onLocationChange: (location: LocationDto | null) => void;
    onSalePointChange: (salePoint: SalePointDto | null) => void;
    onVisibilityChange: (checked: boolean) => void;
    visibility: boolean;
    error: { location?: string; salePoint?: string; server?: string; };
    location: LocationDto | null;
    columnMode?: boolean;
    salePoint?: SalePointDto | null;
};

const VISIBILITY_FIELD = {
    name: 'visibility',
    label: 'Включить видимость',
};

const PromoCampaignVisibilitySettingInput: React.FC<PromoCampaignVisibilitySettingInputProps> = ({
    error,
    onLocationChange,
    onSalePointChange,
    visibility,
    onVisibilityChange,
    location,
    salePoint,
    columnMode,
}) => (
    <div className={cn(styles.wrapper, { [styles.columnWrapper]: columnMode })}>
        <div className={cn(styles.form, { [styles.columnForm]: columnMode })}>
            <AutocompleteLocationAndSalePoint
                onLocationChange={onLocationChange}
                onSalePointChange={onSalePointChange}
                locationLabelClassNames="required"
                locationId={location?.id}
                initialLocationValue={getStringOptionValue(location || undefined)}
                initialSalePointValue={salePoint?.name ?? ''}
                error={error}
                columnMode={columnMode}
            />
            <div className={styles.visibility}>
                <label htmlFor={VISIBILITY_FIELD.name} className={styles.label}>
                    {VISIBILITY_FIELD.label}
                </label>
                <Switch
                    className={cn(styles.switcher, { [styles.switch]: visibility })}
                    checked={visibility}
                    onChange={onVisibilityChange}
                    // id={ VISIBILITY_FIELD.name }
                />
            </div>
        </div>
        {error.server && <div className={styles.error}>{error.server}</div>}
    </div>
);

export default PromoCampaignVisibilitySettingInput;
