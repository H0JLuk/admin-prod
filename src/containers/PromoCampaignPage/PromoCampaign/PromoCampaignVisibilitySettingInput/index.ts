import PromoCampaignVisibilitySettingInput from './PromoCampaignVisibilitySettingInput';

export type { PromoCampaignVisibilitySettingInputProps } from './PromoCampaignVisibilitySettingInput';

export default PromoCampaignVisibilitySettingInput;
