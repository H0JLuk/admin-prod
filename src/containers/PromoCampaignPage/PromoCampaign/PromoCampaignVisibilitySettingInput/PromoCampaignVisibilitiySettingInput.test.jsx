import React from 'react';
import PromoCampaignVisibilitySettingInput from './PromoCampaignVisibilitySettingInput';
import { shallow } from 'enzyme';

describe('<PromoCampaignVisibilitySettingInput /> tests', () => {
    const props = {
        error: {
            location: '',
            salePoint: '',
            server: '',
        },
        onLocationChange: jest.fn(),
        onSalePointChange: jest.fn(),
        visibility: true,
        onVisibilityChange: jest.fn(),
        location: {
            id: 2046,
            name: 'Ростов',
            description: null,
            startDate: '2020-11-01',
            endDate: null,
            deleted: false,
            type: {
                id: 11,
                name: 'Город',
                description: null,
                startDate: '2020-01-01',
                endDate: null,
                priority: 6,
                deleted: false
            },
            parentName: 'Ярославская область'
        },
        salePoint: '',
        columnMode: true,
    };

    it('PromoCampaignVisibilitySettingInput snapshot', () => {
        const VisibilityInput = shallow(<PromoCampaignVisibilitySettingInput { ...props } />);
        expect(VisibilityInput.html()).toMatchSnapshot();
    });

    it('should render error server status', () => {
        const newProps = {
            ...props,
            error: {
                server: 'test'
            }
        };

        const VisibilityInput = shallow(<PromoCampaignVisibilitySettingInput { ...newProps } />);
        expect(VisibilityInput.find('.error').text()).toEqual(newProps.error.server);
    });

    it('should render visibility label', () => {
        const VisibilityInput = shallow(<PromoCampaignVisibilitySettingInput { ...props } />);
        expect(VisibilityInput.find('.label').text()).toEqual('Включить видимость');
    });

    it('should change visibility', () => {
        const VisibilityInput = shallow(<PromoCampaignVisibilitySettingInput { ...props } />);
        VisibilityInput.setProps({ ...props, visibility: false });
        expect(VisibilityInput.find('Switch').prop('checked')).toEqual(!props.visibility);
    });
});
