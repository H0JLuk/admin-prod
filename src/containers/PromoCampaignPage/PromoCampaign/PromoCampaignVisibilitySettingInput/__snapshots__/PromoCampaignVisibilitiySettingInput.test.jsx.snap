// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCampaignVisibilitySettingInput /> tests PromoCampaignVisibilitySettingInput snapshot 1`] = `"<div class=\\"wrapper columnWrapper\\"><div class=\\"form columnForm\\"><div class=\\"formColumn\\"><div class=\\"formLabel required\\"><label for=\\"rc_select_0\\">Локация</label></div><div class=\\"inputWrapper inputWrapperColumn\\"><div class=\\"ant-select ant-select-auto-complete autocompleteInput autocompleteInputColumn hideSuffix ant-select-single ant-select-allow-clear ant-select-customize-input ant-select-show-search\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><span class=\\"ant-input-affix-wrapper ant-select-selection-search-input\\"><input type=\\"search\\" autoComplete=\\"off\\" placeholder=\\"Область, город...\\" name=\\"location\\" autofocus=\\"\\" class=\\"ant-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"undefined_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"undefined_list\\" aria-activedescendant=\\"undefined_list_0\\" value=\\"Ярославская область, Ростов\\"/><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"search\\" class=\\"anticon anticon-search suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"search\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></span></span></span></span></div><span class=\\"ant-select-clear\\" style=\\"user-select:none;-webkit-user-select:none\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><svg>cross.svg</svg></span></div></div></div><div class=\\"formColumn salePointColumn\\"><div class=\\"formLabel\\"><label for=\\"rc_select_1\\">Точка продажи</label></div><div><div class=\\"ant-select ant-select-auto-complete autocompleteInput autocompleteInputColumn ant-select-single ant-select-allow-clear ant-select-customize-input ant-select-show-search\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><span class=\\"ant-input-affix-wrapper ant-select-selection-search-input\\"><input type=\\"search\\" autoComplete=\\"off\\" placeholder=\\"Отделение ВСП\\" name=\\"salePoint\\" class=\\"ant-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"undefined_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"undefined_list\\" aria-activedescendant=\\"undefined_list_0\\" value=\\"\\"/><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"search\\" class=\\"anticon anticon-search suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"search\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></span></span></span></span><span class=\\"ant-select-selection-placeholder\\"></span></div></div></div></div><div class=\\"visibility\\"><label for=\\"visibility\\" class=\\"label\\">Включить видимость</label><button id=\\"visibility\\" type=\\"button\\" role=\\"switch\\" aria-checked=\\"true\\" class=\\"ant-switch switcher switch ant-switch-checked\\"><div class=\\"ant-switch-handle\\"></div><span class=\\"ant-switch-inner\\"></span></button></div></div></div>"`;
